extension-style/src/main/java/com/mapbox/maps/extension/style/sources/Source.kt

```diff
@@ -93,6 +113,14 @@ abstract class Source(
   private fun updateProperty(property: PropertyValue<*>, throwRuntimeException: Boolean = true) {
     delegate?.let { styleDelegate ->
       try {
+        if (this is GeoJsonSource) {
+          val isNativeStyleValid = styleObjectIsValidMethod?.invoke(styleDelegate)
+          // explicitly reset native reference and return
+          if (isNativeStyleValid == false) {
+            delegate = null
+            return@let
+          }
+        }
```

Comment:
Consider the update property can be called very frequently, I'd suggest avoid using reflection. Introducing a new method in style interface is less harmful imo.

Reply:
as discussed with @yunikkk a bit and looking into e.g. [this](https://www.optaplanner.org/blog/2018/01/09/JavaReflectionButMuchFaster.html#:~:text=Adding%20setAccessible(true)%20call%20makes,takes%205.5%20nanoseconds%20per%20call.&text=Reflection%20is%20104%25%20slower%20than,takes%20longer%20to%20warm%20up) it seems that when method is found - calling it using reflection is pretty fast but in general I tend to agree, reflection is evil. I'll try extending an interface as proposed by you in https://github.com/mapbox/mapbox-maps-android/pull/1230#discussion_r830917178

Reply:
reflection now removed